src/movies.js

```diff
@@ -1,25 +1,48 @@
+
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    const allDirectors = moviesArray.map(movie => movie.director)
+    return allDirectors
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+    const dramaMovies = moviesArray.filter(movie => movie.director === "Steven Spielberg" && movie.genre.includes("Drama"))
+    // const dramaMovies = spielbergMovies.filter(spielberg => spielberg.genre.includes("Drama"))
+    return dramaMovies.length
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+    const averageMovieScore = moviesArray.reduce((acc, current) => acc + current.score, 0) / moviesArray.length
+    return averageMovieScore
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) { 
+    const dramaMovies = moviesArray.filter(movie => movie.genre.includes("Drama"))
+    return dramaMovies.length
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear(moviesArray) {}
+function orderByYear(moviesArray) {
+    const sortByYear = [...moviesArray].sort((a, b) => a.year - b.year)
```

Comment:
nice. again the intermediate step with the variable name is not necessarily needed. Only have it there if you the extra line. 
otherwise just return.
```js
return [...moviesArray].sort((a, b) => a.year - b.year)
```

Reply:
however I see, that this is not completed yet right? 
it missed the case where a.year === b.year and we want to compare a.title to b.title instead. 
A little hint -> don't forget to boolean convert the string comparison, maybe like so?
```js
a.title < b.title ? -1 : 1)
```
so - spoken in algorithms -> 
IF a.year and b.year are equal, sort by titles, i.e. return -1 if a.title comes first and 1 if b.title comes first, ELSE sort by years, like you already do.